// @flipbyte/formik-json/es/registry.js
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

export var CONTAINER = 'container';
export var FIELD = 'field';
export var TEMPLATE = 'template';

var Registry = function () {
    function Registry() {
        _classCallCheck(this, Registry);

        this.mapping = {};
    }

    Registry.prototype.get = function get(name) {
        var o = this.mapping[name];
        if (o == null) throw new Error('No object registered for: ' + name);
        return o;
    };

    Registry.prototype.register = function register(name, o) {
        this.mapping[name] = o;
    };

    return Registry;
}();

export var fields = new Registry();
export var registerField = fields.register.bind(fields);

export var containers = new Registry();
export var registerContainer = containers.register.bind(containers);

export var templates = new Registry();
export var registerTemplate = templates.register.bind(templates);

export default Registry;
